e2e/dractions/actions.go

```diff
@@ -79,9 +80,11 @@ func EnableProtection(ctx types.Context) error {
 		return err
 	}
 
-	// For volsync based replication we must create the cluster namespaces with special annotation.
-	if err := util.CreateNamespaceAndAddAnnotation(ctx.Env(), ctx.AppNamespace(), log); err != nil {
-		return err
+	// For Kubernetes, creates namespaces and adds annotation on both DR clusters for volsync based replication.
+	if config.Distro == e2econfig.DistroK8s {
+		if err := util.CreateNamespaceAndAddAnnotation(ctx.Env(), appNamespace, log); err != nil {
+			return err
+		}
```

Comment:
Instead of silencing the linter, do this in a helper function.

Reply:
Instead of helper function for this,  planning to fix #1827 soon to remove funlen and complexity. wdyt?

Reply:
We need to make progress without making things worse. Adding a helper function for this is needed regardless of moving the rety logic. The details of the flows on specific distros should not be in the main protect flow anyway.

Reply:
added an helper function in dractions/actions.go called createNamespaceAndAnnotation